vm/devices/support/fs/lxutil/src/windows/symlink.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use pal::windows::UnicodeString;
+use pal::windows::UnicodeStringRef;
+use windows::Wdk::Storage::FileSystem;
+use windows::Win32::System::SystemServices as W32Ss;
+
+/// Get the symlink substitute name and flags from the reparse data.
+fn get_substitute_name(
+    reparse: &FileSystem::REPARSE_DATA_BUFFER,
+) -> lx::Result<(UnicodeStringRef<'_>, u32)> {
+    // SAFETY: Caller guarantees that the reparse data buffer is well-formed.
```

Comment:
If there are invariants the caller must guarantee for safety, then the function must be marked `unsafe`.